### PR TITLE
Remove What's new  fancy example

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Remove 
https://matplotlib.org/gallery/pyplots/whats_new_98_4_fancy.html

which is redundant with
https://matplotlib.org/gallery/shapes_and_collections/fancybox_demo.html
https://matplotlib.org/gallery/text_labels_and_annotations/fancyarrow_demo.html

and the latter have already better styling.

### CODE REVIEW

**Comment by anntzer** [comment]
that'll break the links at https://matplotlib.org/users/prev_whats_new/whats_new_0.98.4.html#fancy-annotations-and-arrows I think... (probably can be fixed using `.. plot:` instead in that file)

**Comment by timhoffm** [comment]
Links fixed. Not by using `.. plot:` because embedding all the code seems a bit much and would be a duplication. I'm just linking to the remaining fancybox_demo and fancyarrow_demo.

**Comment by anntzer** [comment]
My preference would be to leave the 0.98.4 whatsnew page appear as it does (so with a ..plot::).  The duplication is not really a problem; in my view the whatsnew is just a snapshot of how matplotlib was when 0.98.4 was released.

**Comment by timhoffm** [comment]
@anntzer I've added a `.. plot::` but it's quite large, and elements are overlapping and I don't know how to size it properly.

The change is in a second commit, to be able to undo just in case. Squash if you like.

**Comment by anntzer** [comment]
doc build is failing, probably needs a rebase?

**Comment by timhoffm** [comment]
ping @anntzer 
https://22316-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/users/prev_whats_new/whats_new_0.98.4.html

File: doc/users/prev_whats_new/whats_new_0.98.4.rst [hunk 1]
```diff
@@ -49,18 +49,66 @@ boxes and connectors in annotations.  See
...
-   :scale: 50
+.. plot::
 
-   Whats New 98 4 Fancy
+    import matplotlib.patches as mpatch
+    import matplotlib.pyplot as plt
 
-.. _psd-amplitude:
+    figheight = 8
+    fig = plt.figure(figsize=(9, figheight), dpi=80)
```
Comment (anntzer) [inline doc/users/prev_whats_new/whats_new_0.98.4.rst]: You can perhaps change the size here?  Dunno, but perhaps worth trying?

Comment (timhoffm) [reply doc/users/prev_whats_new/whats_new_0.98.4.rst]: I've reduced the size, but that needed more parameter changes. Now, it's a smaller version as desired. Drawbacks:

- The result is left aligned whereas all other plots are centered.
- This preserves the bad layout of the original example.

I don't care too much if we take this or my first proposal with the links to up-to-date examples.

Comment (anntzer) [reply doc/users/prev_whats_new/whats_new_0.98.4.rst]: I wouldn't worry about the layout, the changelog entry is what it is.
The left-alignment wouldn't be a problem if we also move the other two examples  to inline in the whatsnew :p

**Comment by anntzer** [comment]
I'll let you self-merge with or without squashing first (https://github.com/matplotlib/matplotlib/pull/13685#issuecomment-501924752), up to you.

**Pull request merged**